saleor/webhook/transport/shipping.py

```diff
@@ -209,36 +146,31 @@ def get_excluded_shipping_data(
 
 def get_excluded_shipping_methods_from_response(
     response_data: dict,
-) -> list[dict]:
+    webhook: "Webhook",
+) -> list[ExcludedShippingMethodSchema]:
     excluded_methods = []
-    for method_data in response_data.get("excluded_methods", []):
-        try:
-            type_name, method_id = from_global_id_or_error(method_data["id"])
-            if type_name not in (APP_ID_PREFIX, str(ShippingMethod)):
-                logger.warning(
-                    "Invalid type received. Expected ShippingMethod, got %s", type_name
-                )
-                continue
-
-        except (KeyError, ValueError, TypeError, GraphQLError) as e:
-            logger.warning("Malformed ShippingMethod id was provided: %s", e)
-            continue
-        excluded_methods.append(
-            {"id": method_id, "reason": method_data.get("reason", "")}
+    try:
+        excluded_methods.extend(
```

Comment:
[Sugestions]
Maybe worth to split it into two lines? Right now it looks like a lot of things happens here 😅 

```python
        parsed_app_response = FilterShippingMethodsSchema(response_data) 
        excluded_methods.extend(
            parsed_app_response.excluded_methods
        )
```

Reply:
Done